app/controllers/countries_controller.rb

```diff
@@ -5,6 +5,7 @@ def show
     @country = Country.find(params[:id])
 
     @news_feeds = NewsPresenter.new(@country).find_news
+    @travel_requirements = TravelNotesPresenter.new(@country).find_advisories
```

Comment:
@vaidashi Nice attempt with the presenters 👍 

One of the goals with using presenter objects is to allow your view to only have one instance variable sent to it. Your presenter should define methods that return the results of the other instance variables being set in your controller. So you might end up with this instead:

```ruby
def show
  @presenter = CountryPresenter.new(Country.find(params[:id])
end
```

You'd then define your methods in the presenter something like this

```ruby
class CountryPresenter
  def news
    @news ||= raw_news.map do |raw_story|
      News.new(raw_story)
    end
  end
 
  private
  
    def raw_news
      @raw_news ||= NewsFinder.new(@country).recent_news_call
    end
end
```

In your view you would then call `@presenter.news` to get the individual news items. You'd then add a `travel_requirements` method to the presenter class similar to what I did above for `news`. Also make a note of the memoization I added. This will cut down on multiple API calls in the future.